pkg/user/users_test.go

```diff
@@ -0,0 +1,606 @@
+	roles, err := service.ReadAllRoles(ctx)
+	if expect.NoError(err) {
+		expect.Equal(roles, allRoles)
+	}
+}
+
+func TestReadUsersByRole(t *testing.T) {
+	expect := assert.New(t)
+	usersByRole, err := service.ReadUsersByRole(ctx, "vp", false, 1, "")
+	if expect.NoError(err) && expect.NotEmpty(usersByRole) {
+		expect.Subset(knownUsers, usersByRole)
+	}
+}
+
+func TestReadUsersByRoleAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	usersByRole, err := service.ReadUsersByRoleAsJSON(ctx, "engineer", false, 1, "")
+	if expect.NoError(err) && expect.NotEmpty(usersByRole) {
+		expect.Contains(string(usersByRole), u20.Roles[0])
+	}
+}
+
+func TestReadAllUsersByRole(t *testing.T) {
+	expect := assert.New(t)
+	usersByRole, err := service.ReadAllUsersByRole(ctx, "assistant")
+	if expect.NoError(err) && expect.NotEmpty(usersByRole) {
+		var numOfUsers []int
+		for i := range usersByRole {
+			numOfUsers = append(numOfUsers, i)
+		}
+		expect.GreaterOrEqual(len(numOfUsers), 4)
+	}
+}
+
+func TestReadAllUsersByRoleNegative(t *testing.T) {
+	expect := assert.New(t)
+	noUsers, err := service.ReadAllUsersByOrgID(ctx, "ceo")
+	if expect.NoError(err) {
+		expect.Empty(noUsers)
+	}
+}
+
+func TestReadAllUsersByRoleAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	users, err := service.ReadAllUsersByRoleAsJSON(ctx, "admin")
+	if expect.NoError(err) && expect.NotEmpty(users) {
+		expect.Contains(string(users), u11.Roles[0])
+	}
+}
+
+//------------------------------------------------------------------------------
+// Users by Status
+//------------------------------------------------------------------------------
+
+func TestReadStatuses(t *testing.T) {
+	expect := assert.New(t)
+	// Read all statuses
+	statuses, err := service.ReadStatuses(ctx, false, 3, "-")
+	if expect.NoError(err) {
+		expect.Equal(statuses, allStatuses)
+	}
+}
+
+func TestReadAllStatuses(t *testing.T) {
+	expect := assert.New(t)
+	// Read all statuses
+	statuses, err := service.ReadAllStatuses(ctx)
+	if expect.NoError(err) {
+		expect.Equal(statuses, allStatuses)
+	}
+}
+
+func TestReadUsersByStatus(t *testing.T) {
+	expect := assert.New(t)
+	checkUsers, err := service.ReadUsersByStatus(ctx, "DISABLED", false, 1, "")
+	if expect.NoError(err) && expect.NotEmpty(checkUsers) {
+		expect.Equal(u50, checkUsers[0])
+	}
+}
+
+func TestReadUsersByStatusAsJSON(t *testing.T) {
+	expect := assert.New(t)
+	checkUsers, err := service.ReadUsersByStatusAsJSON(ctx, "DISABLED", false, 1, "")
+	if expect.NoError(err) {
+		expect.Contains(string(checkUsers), u50.ID)
+	}
+}
+
+func TestReadAllUsersByStatus(t *testing.T) {
+	expect := assert.New(t)
+	checkUsers, err := service.ReadAllUsersByStatus(ctx, "ENABLED")
+	if expect.NoError(err) && expect.NotEmpty(checkUsers) {
+		var numOfUsers []int
+		var usersIDs []string
+		for i, v := range checkUsers {
+			numOfUsers = append(numOfUsers, i)
+			usersIDs = append(usersIDs, v.ID)
+		}
+		expect.GreaterOrEqual(len(numOfUsers), 4)
+		expect.Subset(knownIDs, usersIDs)
```

Comment:
I think these two arguments should be reversed. `userIDs` might contain the ID of the CRUD user.